src/Atlas2DSlice.ts

```diff
@@ -60,16 +60,32 @@ export default class Atlas2DSlice implements VolumeRenderImpl {
 
     this.geometryTransformNode.add(this.boxHelper, this.geometryMesh);
 
-    this.setUniform("ATLAS_X", volume.imageInfo.cols);
-    this.setUniform("ATLAS_Y", volume.imageInfo.rows);
-    this.setUniform("textureRes", new Vector2(volume.imageInfo.atlas_width, volume.imageInfo.atlas_height));
-    this.setUniform("SLICES", volume.z);
-    this.setUniform("Z_SLICE", Math.floor(volume.z / 2));
-
-    this.channelData = new FusedChannelData(volume.imageInfo.atlas_width, volume.imageInfo.atlas_height);
+    this.setUniform("Z_SLICE", Math.floor((volume.imageInfo.vol_size_z || volume.z) / 2));
+    this.initChannelData();
+    this.updateVolumeDimensions();
+    this.settings = settings;
     this.updateSettings(settings, SettingsFlags.ALL);
   }
 
+  private initChannelData() {
+    /* eslint-disable-next-line @typescript-eslint/naming-convention */
+    const { cols, rows, tile_width, tile_height } = this.volume.imageInfo;
+    const atlasWidth = tile_width * cols;
+    const atlasHeight = tile_height * rows;
+
+    this.setUniform("ATLAS_X", cols);
+    this.setUniform("ATLAS_Y", rows);
+    this.setUniform("textureRes", new Vector2(atlasWidth, atlasHeight));
+    this.setUniform("SLICES", this.volume.imageInfo.vol_size_z || this.volume.z);
```

Comment:
do we need to worry about distinguishing vol_size_z === 0 versus undefined here?

Reply:
I thought about this - arguably yes, but if volume size is 0 you probably have bigger problems and maybe ought to fall back on data size